test/e2e/storage/testsuites/provisioning.go

```diff
@@ -281,11 +281,17 @@ func TestDynamicProvisioning(t StorageClassTest, client clientset.Interface, cla
 			command += fmt.Sprintf(" && ( mount | grep 'on /mnt/test' | awk '{print $6}' | sed 's/^(/,/; s/)$/,/' | grep -q ,%s, )", option)
 		}
 		command += " || (mount | grep 'on /mnt/test'; false)"
-		runInPodWithVolume(client, claim.Namespace, claim.Name, t.NodeName, command)
+		RunInPodWithVolume(client, claim.Namespace, claim.Name, t.NodeName, command)
 
 		By("checking the created volume is readable and retains data")
-		runInPodWithVolume(client, claim.Namespace, claim.Name, t.NodeName, "grep 'hello world' /mnt/test/data")
+		RunInPodWithVolume(client, claim.Namespace, claim.Name, t.NodeName, "grep 'hello world' /mnt/test/data")
 	}
+
+	if claim.Spec.DataSource != nil {
+		By("checking the created volume whether has the pre-populated data")
+		RunInPodWithVolume(client, claim.Namespace, claim.Name, t.NodeName, "grep 'hello world' /mnt/test/initialData")
```

Comment:
It might be better to populate the data with something unique like the test namespace.  That way we know it's the correct data from this test run.

Reply:
Done